org.civicrm.hremergency/tests/phpunit/api/v3/Contact/DeleteemergencycontactTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use Tests\CiviHR\HREmergency\Fabricator\EmergencyContactFabricator;
+
+/**
+ * @group headless
+ */
+class api_v3_Contact_DeleteEmergencyContactTest extends \PHPUnit_Framework_TestCase
+  implements HeadlessInterface, TransactionalInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  public function testDeletingNonExistentContact() {
+    $message = "Could not find emergency contact with ID '0'";
+    $this->setExpectedException(CiviCRM_API3_Exception::class, $message);
+    civicrm_api3('Contact', 'deleteemergencycontact', ['id' => 0]);
+  }
+
+  public function testDeletingWithoutPermission() {
+    $contacts = civicrm_api3('Contact', 'get');
```

Comment:
We shouldn't assume there will be contacts available in the database. Please use the ContactFabricator to create what you need

Reply:
This was removed since everything is mocked now and doesn't require real contacts